go/base-arm/Dockerfile.tmpl

```diff
@@ -38,11 +38,19 @@ RUN \
         && rm -rf /var/lib/apt/lists/*
 
 ARG GOLANG_VERSION=1.23.5
+{{- if eq .FIPS "true"}}
+ARG GOLANG_DOWNLOAD_URL=https://aka.ms/golang/release/latest/go$GOLANG_VERSION-1.linux-arm64.tar.gz
+# Use a different arg name for microsoft/go sha so it can be handled seperately from the regular golang sha
+ARG MSFT_DOWNLOAD_SHA256=f0a601089c1a523f72f645b5d47cc6582dae495f70b41da614fe193d1931e649
```

Comment:
Is this really necessary? I don't see it used except when setting `DOWNLOAD_SHA256` to it.

Reply:
`GOLANG_DOWNLOAD_URL` and `MSFT_DOWNLOAD_SHA256` get assigned to `DOWNLOAD_SHA256` so it can be used to verify the download
They are different variables so `bump-go-release-version.sh` can replace the upstream golang sha, and the microsoft/go sha independently